python/tests/test_ld_matrix.py

```diff
@@ -699,6 +735,210 @@ def r2_summary_func(
             result[k] = (D * D) / denom
 
 
+def D_summary_func(
+    state_dim: int, state: np.ndarray, result: np.ndarray, params: Dict[str, Any]
+) -> None:
+    """Summary function for the D statistic. We first compute the proportion of
+    AB, A, and B haplotypes, then we compute the D statistic, storing the outputs
+    in the result vector, one entry per sample set.
+
+    :param state_dim: Number of sample sets.
+    :param state: Counts of 3 haplotype configurations for each sample set.
+    :param result: Vector of length state_dim to store the results in.
+    :param params: Parameters for the summary function.
+    """
```

Comment:
Are these docstrings too much? I'm not sure how much they're really adding to the understanding of this code.

Reply:
Could just document one of the summary funcs, and leave the rest clear? It is helpful documenting the first one here